### ISSUE TITLE
Useless quaternions in multiplication

### ISSUE THREAD

**Issue by stla** [AudioSceneDescriptionFormat/splines#4]
Hello,

  Here:

```python
def _natural_control_quaternion(quaternions):
    """Return second control quaternion given the other three."""
    outer, inner_control, inner = quaternions
    return (
        (inner_control * inner.inverse()) *
        (inner * outer.inverse())
    )**(1 / 2) * outer
```

you do the multiplication `(inner_control * inner.inverse()) * (inner * outer.inverse())`. But the quaternions multiplication is associative, so this multiplication is the same as `inner_control * outer.inverse()`.

**Comment by mgeier** [comment]
Thanks for reporting this!

TBH, I don't really remember how I derived this equation, I guess there was a fair bit of trial and error involved.

In the last few days I've properly derived the equation for the Euclidean case (e15e018ea9af9f36010a0c540f1e87dacfe9597a, rendered: https://splines.readthedocs.io/en/latest/euclidean/end-conditions-natural.html#Bézier-Control-Points), which leads to the simplified equation that you suggested (c8707fe06bc07ce6182a7fd0d8bcf2c244bd5c9b, rendered: https://splines.readthedocs.io/en/latest/rotation/end-conditions-natural.html). The implementation now looks like this:

https://github.com/AudioSceneDescriptionFormat/splines/blob/c8707fe06bc07ce6182a7fd0d8bcf2c244bd5c9b/src/splines/quaternion.py#L611-L622

I've also fixed this in the Rust implementation: https://github.com/AudioSceneDescriptionFormat/asdfspline-rust/pull/11.
